source/input.cpp

```diff
@@ -478,23 +517,47 @@ void reshade::input::block_keyboard_input(bool enable)
 	_block_keyboard = enable;
 }
 
-bool is_blocking_mouse_input()
+bool is_immobilizing_cursor(reshade::input::window_handle target = reshade::input::GlobalQueue)
 {
-	const std::shared_lock<std::shared_mutex> lock(s_windows_mutex);
+	if (target != reshade::input::GlobalQueue)
+	{
+		const std::shared_lock<std::shared_mutex> lock(s_windows_mutex);
 
-	const auto predicate = [](const std::pair<HWND, std::weak_ptr<reshade::input>> &input_window) {
-		return !input_window.second.expired() && input_window.second.lock()->is_blocking_mouse_input();
-	};
-	return std::any_of(s_windows.cbegin(), s_windows.cend(), predicate);
+		const auto predicate = [&](const std::pair<HWND, std::weak_ptr<reshade::input>> &input_window) {
+			return !input_window.second.expired() && (target == reshade::input::AnyWindow || input_window.first == target) && input_window.second.lock()->is_immobilizing_cursor();
+		};
+		return std::any_of(s_windows.cbegin(), s_windows.cend(), predicate);
+	}
+
+	return s_immobilizing_global_cursor_move.load();
 }
-bool is_blocking_keyboard_input()
+bool is_blocking_mouse_input(reshade::input::window_handle target = reshade::input::GlobalQueue)
 {
-	const std::shared_lock<std::shared_mutex> lock(s_windows_mutex);
+	if (target != reshade::input::GlobalQueue)
```

Comment:
I'm not sure I understand the need to distinguish `GlobalQueue` and `AnyWindow`. It seems to me that in both cases it would be sufficient to just check the global atomic bool?

Reply:
Yeah, that's a good point.